### ISSUE TITLE
fossology can't unpack .tar.xz archives

### ISSUE THREAD

**Issue by dankegel** [fossology/fossology#495]
Analyzing
http://ftp.gnu.org/gnu/bison/bison-3.0.2.tar.xz
fails; fossology reports "no files".

It kind of looks like src/ununpack/agent needs to learn about .tar.xz files.

Given that modern tar can figure out how to unpack things on its own, maybe something like this would do (completely untested):
```
--- a/src/ununpack/agent/utils.c
+++ b/src/ununpack/agent/utils.c
@@ -869,7 +869,7 @@ int FindCmd (char *Filename)
       strcpy(TypeBuf,"application/x-zip");
     }
   } /* if was x-exe */
-  else if (strstr(Type, "application/x-tar"))
+  else if (strstr(Type, "application/x-tar") || strstr(Type, "application/x-xz"))
   {
     if (RunCommand("tar","-tf",Filename,">/dev/null 2>&1",NULL,NULL) != 0)
       return(-1); /* bad tar! (Yes, they do happen) */
```

This is somewhat more important now that some projects are *only* publishing source tarballs in .xz form, e.g.
ftp://ftp.gnome.org/pub/GNOME/sources/GConf/3.2
contains .tar.xz but no other tarball formats for things published in the last three years.

**Comment by dankegel** [comment]
I just tested that patch on 2.6.2, and it seems to work.

**Comment by bobgob** [comment]
Hmmm, we do uncompress .xz's.  This works in 2.6.2 even without your patch.  I'm thinking that maybe we missed a dependency.  What OS are you running?
In the mean time I don't see how this would hurt so I committed your fix on f3bdc6e

**Comment by dankegel** [comment]
Now that I actually look at the source again, I was clearly asleep when I wrote that.
No idea what the problem really was, or why it went away.

I installed three times, all on Ubuntu 14.04 or so.  The instance I care about was installed in a
clean virtual machine, created with 
   lxc-create -n fossology -t ubuntu -- -r trusty
but I'm not sure that one ever suffered from the problem.

I'll try to have my act more together next time I post a patch.

In the meantime, it'd probably be better if you reverted it.
